captum/insights/attr_vis/app.py

```diff
@@ -199,6 +200,7 @@ class scores.
         self._outputs: List[VisualizationOutput] = []
         self._config = FilterConfig(prediction="all", classes=[], num_examples=4)
         self._dataset_iter = iter(dataset)
```

Comment:
wouldn't your PR be simpler to construct this as `cycle(iter(dataset))` since it seems you're effectively reimplementing this?

Reply:
It would be much simpler that way, but my concern is about memory usage. Per the [docs](https://docs.python.org/3/library/itertools.html#itertools.cycle), it seems that cycle
> may require significant auxiliary storage (depending on the length of the iterable)

Which suggests to me that it stores the entire thing in an array and then loops around. If the dataset is huge, then I think this could be an issue, because every batch would get stored in memory. That's why I made the cache only store a few batches instead, rather than using a cycle over the entire dataset. 

However, if you don't think the memory is an issue, I could do it this way instead. It would probably only be an issue if somebody requested a _lot_ of attributions.

Reply:
Makes sense. Thanks for clarifying. Probably best to assume memory will be an issue.